### PR TITLE
Add unit tests for `ZCL_ABAPGIT_GUI_PAGE_SETT_REMO->VALIDATE_FORM`

### PR DESCRIPTION
**Pull request by christianguenter2**
- no functional changes
- preparation for refactoring ZCL_ABAPGIT_GUI_PAGE_SETT_REMO->VALIDATE_FORM and others
- add new interface ZIF_ABAPGIT_GIT_TRANSPORT
- allows mocking ZIF_ABAPGIT_GIT_TRANSPORT via ZCL_ABAPGIT_GIT_INJECTOR

### CODE REVIEW

**Comment by christianguenter2** [comment]
Hm, I don't know how to fix test unit-tests error. As `zif_abapgit_frontend_services` shouldn't it work?

**Comment by larshp** [comment]
try moving the injection to the top of the method

line 204 calls the frontend services before its mocked

image

**Comment by christianguenter2** [comment]
Thx. Any idea what's wrong with the integration test? Is it possible to run the integration tests inside GH codespaces?

**Comment by larshp** [comment]
try cleaning up the mocks in `teardown`

**Comment by christianguenter2** [comment]
Hm, didn't help.

File: src/ui/pages/zcl_abapgit_gui_page_sett_remo.clas.testclasses.abap [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+    CREATE OBJECT mo_cut EXPORTING io_repo = mo_repo.
+
+  ENDMETHOD.
+
+
+  METHOD teardown.
+
+    CLEAR:
+      mo_git_transport_mock,
+      mo_frontend_services_mock.
```
Comment (larshp) [inline src/ui/pages/zcl_abapgit_gui_page_sett_remo.clas.testclasses.abap:217]:
```suggestion
      mo_frontend_services_mock.
      
    zcl_abapgit_git_injector=>set_git_transport( mo_git_transport_mock ).
    zcl_abapgit_ui_injector=>set_frontend_services( mo_frontend_services_mock ).
```

Comment (christianguenter2) [reply src/ui/pages/zcl_abapgit_gui_page_sett_remo.clas.testclasses.abap:217]: Thx. Seems I don't understand references enough.

**Comment by larshp** [comment]
looks like this change breaks some other repos? https://github.com/abapGit/abapGit/pull/6712/checks?check_run_id=19827596707

**Comment by christianguenter2** [comment]
Ok, seems that `zcl_abapgit_git_transport=>branches` is a too well established API to change/remove.

**Comment by larshp** [comment]
> Ok, seems that `zcl_abapgit_git_transport=>branches` is a too well established API to change/remove.

its possible, but suggest deferring it to a different PR focusing on just that method

**Review by larshp — APPROVED** [review]
lgtm

**Comment by mbtools** [comment]
works 👍

**Pull request merged**